rdr_service/genomic/genomic_job_components.py

```diff
@@ -745,35 +751,65 @@ def _get_qc_status_from_value(self, aw4_value):
             logging.warning(f'Value from AW4 "{aw4_value}" is not PASS/FAIL.')
             return GenomicQcStatus.UNSET
 
-    def calculate_contamination_category(self, raw_contamination, member):
+    @staticmethod
+    def _participant_has_potentially_clean_samples(session, biobank_id):
+        """Check for any stored sample for the participant that is not contaminated
+        and is a 1ED04, 1ED10, or 1SAL2 test"""
+        query = session.query(BiobankStoredSample).filter(
+            BiobankStoredSample.biobankId == biobank_id
+        ).outerjoin(GenomicSampleContamination).filter(
+            GenomicSampleContamination.id.is_(None),
+            BiobankStoredSample.test.in_(['1ED04', '1ED10', '1SAL2'])
```

Comment:
We may want to check that `BiobankStoredSample.status` is not one of these:
SAMPLE_NOT_RECEIVED = 13
    SAMPLE_NOT_PROCESSED = 14
    ACCESSINGING_ERROR = 15
    LAB_ACCIDENT = 16
    QNS_FOR_PROCESSING = 17
    QUALITY_ISSUE = 18

Reply:
Done! Thanks for catching that.